request_reports.py

```diff
@@ -109,21 +129,48 @@ def getCurrentTimes():
     res = json.loads(response.read())['results']
     print '%d reports received.' % len(res)
 
+    ordered = []
     for report in res:
         priv = bytes_to_int(base64.b64decode(ids[report['id']]))
         data = base64.b64decode(report['payload'])
 
         # the following is all copied from https://github.com/hatomist/openhaystack-python, thanks @hatomist!
         timestamp = bytes_to_int(data[0:4])
-        eph_key = (bytes_to_int(data[6:34]), bytes_to_int(data[34:62]))
-        shared_key = scalar_mult(priv, eph_key)
-        symmetric_key = sha256(int_to_bytes(shared_key[0], 28) + int_to_bytes(1, 4) + data[5:62])
-        decryption_key = symmetric_key[:16]
-        iv = symmetric_key[16:]
-        enc_data = data[62:72]
-        tag = data[72:]
-
-        decrypted = decrypt_AES(enc_data, decryption_key, modes.GCM(iv, tag))
-        res = decode_tag(decrypted)
-        res['timestamp'] = timestamp + 978307200
-        print report['id'] + ': ' + str(res)
+        if timestamp + 978307200 >= startdate:
+          eph_key = (bytes_to_int(data[6:34]), bytes_to_int(data[34:62]))
+          shared_key = scalar_mult(priv, eph_key)
+          symmetric_key = sha256(int_to_bytes(shared_key[0], 28) + int_to_bytes(1, 4) + data[5:62])
+          decryption_key = symmetric_key[:16]
+          iv = symmetric_key[16:]
+          enc_data = data[62:72]
+          tag = data[72:]
+
+          decrypted = decrypt_AES(enc_data, decryption_key, modes.GCM(iv, tag))
+          res = decode_tag(decrypted)
+          res['timestamp'] = timestamp + 978307200
+          res['isodatetime'] = datetime.fromtimestamp(res['timestamp']).isoformat()
+          name=names[report['id']]
+          if args.prefix:
+            res['key'] = name[len(args.prefix)]
```

Comment:
This looks like a missing `:`, I;m slightly redoing this code to make it more compact sorry @olivluca